src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,112 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final double DEFAULT_LOAD_FACTOR = 0.75;
+    private Node<K, V>[] table;
+    private int size = 0;
+
+    MyHashMap() {
+        table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= DEFAULT_LOAD_FACTOR * table.length) {
+            resize();
+        }
+        int hashInputNode = (key == null) ? 0 : key.hashCode();
+        Node<K, V> node = new Node<>(hashInputNode, key, value, null);
+        int index = (hashInputNode == 0) ? 0 : Math.abs(hashInputNode) % table.length;
+        Node<K, V> currentNode = table[index];
+        if (table[index] == null) {
+            table[index] = node;
+            size++;
+        } else {
+            if (currentNode.hash == hashInputNode && (currentNode.key == key
+                    || (currentNode.key != null && currentNode.key.equals(key)))) {
+                currentNode.value = value;
+            } else {
+                for (; ; ) {
+                    if (currentNode.hash == hashInputNode && (currentNode.key == key
+                            || (currentNode.key != null && currentNode.key.equals(key)))) {
+                        currentNode.value = value;
+                        break;
+                    }
+                    if (currentNode.next == null) {
+                        currentNode.next = node;
+                        size++;
+                        break;
+                    }
+                    currentNode = currentNode.next;
+                }
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
```

Comment:
too complex logic, rewrite it following this algorithm:
1. Find index by key and get the first node in this bucket
2. using a while loop, go through this bucket's chain until while node != null and compare keys using Objects.equals. If keys equal return a needed value.
3. If you don't find a needed value, return null

Reply:
optimized